Flaky TestKafkaPlugin

```
Error:  io.trino.plugin.kafka.TestKafkaPlugin.testSslTruststoreMissingFileSpindown -- Time elapsed: 0.220 s <<< FAILURE!
java.lang.AssertionError: 

Expecting throwable message:
  "Configuration errors:

1) Error: Configuration property 'kafka.ssl.endpoint-identification-algorithm' was not used

2) Error: Configuration property 'kafka.ssl.key.password' was not used

3) Error: Configuration property 'kafka.ssl.keystore.location' was not used

4) Error: Configuration property 'kafka.ssl.keystore.password' was not used

5) Error: Configuration property 'kafka.ssl.keystore.type' was not used

6) Error: Configuration property 'kafka.ssl.truststore.location' was not used

7) Error: Configuration property 'kafka.ssl.truststore.password' was not used

8) Error: Configuration property 'kafka.ssl.truststore.type' was not used

8 errors"
to contain:
  "Error: Invalid configuration property kafka.ssl.truststore.location: file does not exist: /not/a/real/path"
but did not.
```

Happened on https://github.com/trinodb/trino/actions/runs/6821919079/job/18553418277

--- a/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaConnectorFactory.java
+++ b/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaConnectorFactory.java
@@ -13,18 +13,21 @@
  */
 package io.trino.plugin.kafka;
 
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Injector;
 import com.google.inject.Module;
 import io.airlift.bootstrap.Bootstrap;
 import io.airlift.json.JsonModule;
 import io.trino.plugin.base.CatalogNameModule;
 import io.trino.plugin.base.TypeDeserializerModule;
+import io.trino.plugin.kafka.security.KafkaSecurityModule;
 import io.trino.spi.NodeManager;
 import io.trino.spi.connector.Connector;
 import io.trino.spi.connector.ConnectorContext;
 import io.trino.spi.connector.ConnectorFactory;
 import io.trino.spi.type.TypeManager;
 
+import java.util.List;
 import java.util.Map;
 
 import static io.trino.plugin.base.Versions.checkStrictSpiVersionMatch;
@@ -33,11 +36,11 @@ import static java.util.Objects.requireNonNull;
 public class KafkaConnectorFactory
         implements ConnectorFactory
 {
-    private final Module extension;
+    private final List<Module> extensions;
 
-    public KafkaConnectorFactory(Module extension)
+    public KafkaConnectorFactory(List<Module> extensions)
     {
-        this.extension = requireNonNull(extension, "extension is null");
+        this.extensions = ImmutableList.copyOf(extensions);
     }
 
     @Override
@@ -54,16 +57,20 @@ public class KafkaConnectorFactory
         checkStrictSpiVersionMatch(context, this);
 
         Bootstrap app = new Bootstrap(
-                new CatalogNameModule(catalogName),
-                new JsonModule(),
-                new TypeDeserializerModule(context.getTypeManager()),
-                new KafkaConnectorModule(context.getTypeManager()),
-                extension,
-                binder -> {
-                    binder.bind(ClassLoader.class).toInstance(KafkaConnectorFactory.class.getClassLoader());
-                    binder.bind(TypeManager.class).toInstance(context.getTypeManager());
-                    binder.bind(NodeManager.class).toInstance(context.getNodeManager());
-                });
+                ImmutableList.<Module>builder()
+                        .add(new CatalogNameModule(catalogName))
+                        .add(new JsonModule())
+                        .add(new TypeDeserializerModule(context.getTypeManager()))
+                        .add(new KafkaConnectorModule(context.getTypeManager()))
+                        .add(new KafkaClientsModule())
+                        .add(new KafkaSecurityModule())
+                        .add(binder -> {
+                            binder.bind(ClassLoader.class).toInstance(KafkaConnectorFactory.class.getClassLoader());
+                            binder.bind(TypeManager.class).toInstance(context.getTypeManager());
+                            binder.bind(NodeManager.class).toInstance(context.getNodeManager());
+                        })
+                        .addAll(extensions)
+                        .build());
 
         Injector injector = app
                 .doNotInitializeLogging()

--- a/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaPlugin.java
+++ b/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaPlugin.java
@@ -14,43 +14,30 @@
 package io.trino.plugin.kafka;
 
 import com.google.common.collect.ImmutableList;
-import com.google.inject.Binder;
 import com.google.inject.Module;
-import io.airlift.configuration.AbstractConfigurationAwareModule;
-import io.trino.plugin.kafka.security.KafkaSecurityModule;
 import io.trino.spi.Plugin;
 import io.trino.spi.connector.ConnectorFactory;
 
-import static java.util.Objects.requireNonNull;
+import java.util.List;
 
 public class KafkaPlugin
         implements Plugin
 {
-    public static final Module DEFAULT_EXTENSION = new AbstractConfigurationAwareModule()
-    {
-        @Override
-        protected void setup(Binder binder)
-        {
-            install(new KafkaClientsModule());
-            install(new KafkaSecurityModule());
-        }
-    };
-
-    private final Module extension;
+    private final List<Module> extensions;
 
     public KafkaPlugin()
     {
-        this(DEFAULT_EXTENSION);
+        this(ImmutableList.of());
     }
 
-    public KafkaPlugin(Module extension)
+    public KafkaPlugin(List<Module> extensions)
     {
-        this.extension = requireNonNull(extension, "extension is null");
+        this.extensions = ImmutableList.copyOf(extensions);
     }
 
     @Override
     public synchronized Iterable<ConnectorFactory> getConnectorFactories()
     {
-        return ImmutableList.of(new KafkaConnectorFactory(extension));
+        return ImmutableList.of(new KafkaConnectorFactory(extensions));
     }
 }

--- a/plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/KafkaQueryRunner.java
+++ b/plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/KafkaQueryRunner.java
@@ -15,7 +15,6 @@ package io.trino.plugin.kafka;
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
-import com.google.inject.Module;
 import com.google.inject.Scopes;
 import io.airlift.json.JsonCodec;
 import io.airlift.log.Logger;
@@ -43,7 +42,6 @@ import java.util.Optional;
 
 import static com.google.common.io.ByteStreams.toByteArray;
 import static io.airlift.configuration.ConditionalModule.conditionalModule;
-import static io.airlift.configuration.ConfigurationAwareModule.combine;
 import static io.airlift.units.Duration.nanosSince;
 import static io.trino.plugin.kafka.util.TestUtils.loadTpchTopicDescription;
 import static java.lang.String.format;
@@ -92,13 +90,6 @@ public final class KafkaQueryRunner
         }
 
         @Override
-        public Builder setExtension(Module extension)
-        {
-            this.extension = requireNonNull(extension, "extension is null");
-            return this;
-        }
-
-        @Override
         public void preInit(DistributedQueryRunner queryRunner)
                 throws Exception
         {
@@ -125,16 +116,14 @@ public final class KafkaQueryRunner
                     .putAll(tpchTopicDescriptions)
                     .putAll(testTopicDescriptions.buildOrThrow())
                     .buildOrThrow();
-            setExtension(combine(
-                    extension,
-                    conditionalModule(
-                            KafkaConfig.class,
-                            kafkaConfig -> kafkaConfig.getTableDescriptionSupplier().equalsIgnoreCase(TEST),
-                            binder -> binder.bind(TableDescriptionSupplier.class)
-                                    .toInstance(new MapBasedTableDescriptionSupplier(topicDescriptions))),
-                    binder -> binder.bind(ContentSchemaProvider.class).to(FileReadContentSchemaProvider.class).in(Scopes.SINGLETON),
-                    new DecoderModule(),
-                    new EncoderModule()));
+            addExtension(conditionalModule(
+                    KafkaConfig.class,
+                    kafkaConfig -> kafkaConfig.getTableDescriptionSupplier().equalsIgnoreCase(TEST),
+                    binder -> binder.bind(TableDescriptionSupplier.class)
+                            .toInstance(new MapBasedTableDescriptionSupplier(topicDescriptions))));
+            addExtension(binder -> binder.bind(ContentSchemaProvider.class).to(FileReadContentSchemaProvider.class).in(Scopes.SINGLETON));
+            addExtension(new DecoderModule());
+            addExtension(new EncoderModule());
             Map<String, String> properties = new HashMap<>(extraKafkaProperties);
             properties.putIfAbsent("kafka.table-description-supplier", TEST);
             setExtraKafkaProperties(properties);

--- a/plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/KafkaQueryRunnerBuilder.java
+++ b/plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/KafkaQueryRunnerBuilder.java
@@ -20,11 +20,12 @@ import io.airlift.log.Logging;
 import io.trino.testing.DistributedQueryRunner;
 import io.trino.testing.kafka.TestingKafka;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import static io.airlift.testing.Closeables.closeAllSuppress;
-import static io.trino.plugin.kafka.KafkaPlugin.DEFAULT_EXTENSION;
 import static io.trino.testing.TestingSession.testSessionBuilder;
 import static java.util.Objects.requireNonNull;
 
@@ -33,7 +34,7 @@ public abstract class KafkaQueryRunnerBuilder
 {
     protected final TestingKafka testingKafka;
     protected Map<String, String> extraKafkaProperties = ImmutableMap.of();
-    protected Module extension = DEFAULT_EXTENSION;
+    private final List<Module> extensions = new ArrayList<>();
     private final String catalogName;
 
     public KafkaQueryRunnerBuilder(TestingKafka testingKafka, String defaultSessionName)
@@ -57,9 +58,10 @@ public abstract class KafkaQueryRunnerBuilder
         return this;
     }
 
-    public KafkaQueryRunnerBuilder setExtension(Module extension)
+    public KafkaQueryRunnerBuilder addExtension(Module extension)
     {
-        this.extension = requireNonNull(extension, "extension is null");
+        requireNonNull(extension, "extension is null");
+        extensions.add(extension);
         return this;
     }
 
@@ -74,7 +76,7 @@ public abstract class KafkaQueryRunnerBuilder
         try {
             testingKafka.start();
             preInit(queryRunner);
-            queryRunner.installPlugin(new KafkaPlugin(extension));
+            queryRunner.installPlugin(new KafkaPlugin(extensions));
             // note: additional copy via ImmutableList so that if fails on nulls
             Map<String, String> kafkaProperties = new HashMap<>(ImmutableMap.copyOf(extraKafkaProperties));
             kafkaProperties.putIfAbsent("kafka.nodes", testingKafka.getConnectString());

